public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard/Configuration.tsx

```diff
@@ -45,15 +45,20 @@ export const Configuration = ({
             <Switch
               {...register('isTimeSelectionEnabled')}
               data-testid={selectors.EnableTimeRangeSwitch}
-              onChange={(e) => onChange('isTimeSelectionEnabled', e.currentTarget.checked)}
+              onChange={(e) => {
+                reportInteraction('grafana_dashboards_public_time_selection_clicked', {
+                  action: e.currentTarget.checked ? 'enable' : 'disable',
+                });
+                onChange('isTimeSelectionEnabled', e.currentTarget.checked);
+              }}
             />
             <Label description="Allow viewers to change time range">Time range picker enabled</Label>
           </Layout>
           <Layout orientation={0} spacing="sm">
             <Switch
               {...register('isAnnotationsEnabled')}
               onChange={(e) => {
-                reportInteraction('grafana_dashboards_annotations_clicked', {
+                reportInteraction('grafana_dashboards_public_annotations_clicked', {
```

Comment:
If the other event was created by mistake, just ping @thanos-karachalios to remove that event to avoid confusion :)